modules/prebid/optimization/config/rules-engine-schema.json

```diff
@@ -0,0 +1,133 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Prebid Optimization Rules Engine Module",
+  "description": "A schema which validates rules engine params",
+  "type": "object",
+  "properties": {
+    "enabled": {
+      "type": "boolean",
+      "description": "pending"
+    },
+    "ruleSets": {
+        "type": "array",
+        "minItems": 1,
+        "items": {
+            "type": "object",
+            "properties": {
+              "stage": {
+                "type": "string",
+                "enum": ["entrypoint", "raw-auction", "processed-auction-request", "bidder-request", "raw-bidder-response", "all-processed-bid-responses", "auction-response"]
+              },
+              "name": {
+                "type": "string",
+                "minLength": 1
+              },
+              "version": {
+                "type": "string"
+              },
+              "modelGroups": {
+                  "type": "array",
+                  "minItems": 1,
+                  "items": {
+                      "type": "object",
+                      "properties": {
+                          "weight": {
+                              "type": "number",
+                              "minimum": 0,
+                              "maximum": 100
+                          },
+                          "analyticsKey": {
+                              "type": "string"
+                          },
+                          "version": {
+                              "type": "string"
+                          },
+                          "schema": {
+                              "type": "array",
+                              "minItems": 1,
+                              "items": {
+                                  "type": "object",
+                                  "properties": {
+                                      "function": {
+                                          "type": "string",
+                                          "enum": ["deviceCountry", "dataCenters", "channel", "eidAvailable", "userFpdAvailable", "fpdAvail", "gppSid", "tcfInScope", "percent", "prebidKey", "domain", "bundle", "deviceType"]
+                                      },
+                                      "args": {
+                                          "type": "array"
+                                      }
+                                  },
+                                  "required": ["function"]
+                              }
+                          },
+                          "default": {
+                              "type": "array"
+                          },
```

Comment:
Default should be an array of objects where each object is a result function similar to `rules[].results`. An empty array is valid.

Reply:
Modified